### PR TITLE
[FIX] account: recompute base amount tax

### PR DESCRIPTION
**Pull request by jpp-odoo**
- In a multi currency company, create an invoice with the domestic
  currency ($);
- Add a product (500$) with taxes;
-  Change the currency to a foreign one (€);
- The product don't change of amount but only of currency (500€);
- Save the invoice, and print it.

Before this commit, the base_amount_tax of the tax wasn't be updated.
Note that the base_amount_tax it should be expressed in domestic
currency. As this field didn't update in this case, the base_amount_tax
is expressing the amount in foreign currency. As this amount is supposed
to be in domestic, for the sake of the reports, it's change in currency
of the invoice (in this case the foreign one), and it's shown in the
report. The amount shown is not correct.

Now, the base_amount_tax is recompute and setted in domestic currency.

opw-2157853

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -339,7 +339,7 @@ def _onchange_currency(self):
                 line.currency_id = new_currency
 
         self.line_ids._onchange_currency()
-        self._recompute_dynamic_lines()
+        self._recompute_dynamic_lines(recompute_all_taxes=True)
```
Comment (smetl) [inline addons/account/models/account_move.py]: You don't want to recompute taxes when changing the date. Instead, you should fix `_compute_invoice_taxes_by_group` that is not re-triggered for some reason.

Comment (jpp-odoo) [reply addons/account/models/account_move.py]: Indeed, I didn't think about the date, sorry by that.
_compute_invoice_taxes_by_group Is correctly triggered, but the tax_base_amount is not correct, for the moment it didn't change, and stays with the values as it was a domestic currency.

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -564,6 +564,10 @@ def _compute_base_line_taxes(base_line):
 
             if not tax_line and not taxes_map_entry['grouping_dict']:
                 continue
+            elif tax_line and recompute_tax_base_amount:
+                tax_line.update({
```
Comment (smetl) [inline addons/account/models/account_move.py]: can you inline this plz? `tax_line.tax_base_amount = tax_base_amount`?

**Comment by jpp-odoo** [comment]
@smetl I have made the changes as requested

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**